test/testkit/mocks/urlprovider/urlprovider.go

```diff
@@ -1,15 +1,15 @@
 package urlprovider
 
 type URLProvider struct {
-	metrics   string
```

Comment:
I also simplified the setup of the URLProvider. Before it was more generic and was capable of storing multiple URLs per pipeline. However, as of now, it only accesses the mock backend URL. Thus, I removed the capability of storing multiple URLs and store only the mock backend URL.